### ISSUE TITLE
Rename jekyll to examples

### ISSUE THREAD

**Issue by anthonydillon** [vanilla-framework/vanilla-framework#803]
I believe it would be better to rename the jekyll directory to examples. As that is what they are.

**Comment by barrymcgee** [comment]
Ideally, I'd like to separate the actual examples from the Jekyll specific files so only `_base`, `_patterns` & `_utilties` would be in an `examples` folder while the rest would remain within `jekyll` - any ideas on how to achieve that?

**Comment by nottrobin** [comment]
The `examples/` folder cannot live above the folder containing `_config.yaml`. And, for this to [work with GitHub pages](https://help.github.com/articles/configuring-a-publishing-source-for-github-pages/#publishing-your-github-pages-site-from-a-docs-folder-on-your-master-branch), `_config.yaml` has to either live in the site root `/`, or in `/docs`. Of course, currently the documentation pages live in `/docs`, which seems to me the natural home for them.

Given these constraints, I can think of 3 basic structures:

1. Everything under `/docs` (https://github.com/ubuntudesign/vanilla-framework/pull/815)

   ``` bash
   vanilla-framework
   ├── docs
   │      ├── _config.yml
   │      ├── _layouts
   │      ├── _plugins
   │      ├── examples/
   │               └── {HTML examples live here}
   │      └──  src/
   │               └── {documentation lives here}
   └── {SCSS & other files live here}
   ```

2. Everything top level (https://github.com/ubuntudesign/vanilla-framework/pull/816)

   ``` bash
   vanilla-framework
   ├── _config.yml
   ├── _layouts
   ├── _plugins
   ├── examples/
   │        └── {HTML examples live here}
   └──  docs/
   │        └── {documentation lives here}
   └── {SCSS & other files live here}
   ```

3. Top level, but keep extra Jekyll files in their own directory, where possible

   ``` bash
   vanilla-framework
   ├── _config.yml
   ├── jekyll/
   │       ├──  _layouts
   │       └──_plugins
   ├── examples/
   │        └── {HTML examples live here}
   └──  docs/
   │        └── {documentation lives here}
   └── {SCSS & other files live here}
   ```

Personally my favourite is structure number 2. I like `/docs` and `/examples` as separate top-level directories, and I think having a `/jekyll` directory just to encapsulate 2 more directories is overkill, plus anyone familiar with Jekyll would be confused by not finding a `_layouts` dir in the usual place.

**Comment by pmahnke** [comment]
I would vote for 3 as having _layouts/ and _plugins/ in root might confuse people that they are part of vanilla, not a jekyll artifact

**Comment by barrymcgee** [comment]
Addressed by #845
